### ISSUE TITLE
Methane Digester Engine - Bug list

### ISSUE THREAD

**Issue by Gunner76th** [GenDeathrow/Hatchery#173]
First I will start off by saying, Yes I did already search for and see that there are bug reports for these listed issues, however said reports are both closed and old. The bugs listed were previously marked as fixed and confirmed as bugs previously and are now back in Hatchery 2.1.4 for Minecraft 1.12.2

The modpack in question is StoneBlock V 1.0.31
Minecraft 1.12.2
Java version 8
Forge 14.23.5.2768

1) Power output only from sides with vents: If you place a block that needs power to run on any side of the engine, that block will receive power. The problem here is that the only 2 sides that will output power beyond 1 block is the sides with the vents. What I means is this: 
     If you connect a leadstone Fluxduct, Ender IO energy conduit, or any other method to the back, front, top, or bottom of the engine and extend that beyond 1 block space before connecting it to a "something" then no power is transmitted along the conduit. If you place ANYTHING (machine, battery, some other energy consuming device) to the conduit at the first connection to the engine, then power is sent down the entire length of the conduit. 
   However, if you connect that same conduit to the sides of the engine where the vents are, it does not matter if you have a "something" on that first block space as power is sent down the entire length of the conduit without issue. Pics below for example:

1) Pic showing Engine on left, Autosieve for testing purpose on right. 
![image](https://user-images.githubusercontent.com/5566872/49581500-2764fe80-f920-11e8-836a-90496bcdc27a.png)

2) Pics showing current internal energy storage of both devices. 
![image](https://user-images.githubusercontent.com/5566872/49581534-419edc80-f920-11e8-817b-b02a100af9b3.png)
![image](https://user-images.githubusercontent.com/5566872/49581539-4794bd80-f920-11e8-8c63-4b9483314566.png)

3) Pis showing Fluxduct, EnderIO Conduit, and XU Energy Transfer Node setups and results when connected to a non-vent side.

![image](https://user-images.githubusercontent.com/5566872/49581644-9c383880-f920-11e8-8a4d-48f4cab24da9.png)
![image](https://user-images.githubusercontent.com/5566872/49581660-a5290a00-f920-11e8-99c4-44e7b2d464ca.png)

![image](https://user-images.githubusercontent.com/5566872/49581742-efaa8680-f920-11e8-9b55-b133a2a9ea3a.png)
![image](https://user-images.githubusercontent.com/5566872/49581748-f46f3a80-f920-11e8-8a13-15a86d688022.png)

![image](https://user-images.githubusercontent.com/5566872/49581805-1e286180-f921-11e8-9e8a-01a1ba3404bd.png)
![image](https://user-images.githubusercontent.com/5566872/49581808-22ed1580-f921-11e8-83bd-0fbae72b745f.png)

4) Pis showing Fluxduct, EnderIO Conduit, and XU Energy Transfer Node setups and results when connected to a vent side.

![image](https://user-images.githubusercontent.com/5566872/49581861-4fa12d00-f921-11e8-921d-34a7fbb2e2d6.png)
![image](https://user-images.githubusercontent.com/5566872/49581872-562fa480-f921-11e8-968d-089dbe76d6a8.png)

![image](https://user-images.githubusercontent.com/5566872/49581890-621b6680-f921-11e8-97ca-477b2d9a1068.png)
![image](https://user-images.githubusercontent.com/5566872/49583295-c04a4880-f925-11e8-9528-f91853259912.png)

![image](https://user-images.githubusercontent.com/5566872/49581943-8a0aca00-f921-11e8-826e-2e066a2d74b7.png)
![image](https://user-images.githubusercontent.com/5566872/49581948-8ecf7e00-f921-11e8-8b10-ea6e5b17d79b.png)

As you can see, if connected to a non-vent side, none of the conduits transmit power. If connected to a vented side, the only conduit that transmits power is the Leadstone one from Thermal Expansion. Through further testing I have also discovered that even the Leadstone one shows an issue in that if only 1 conduit is placed, and an energy consuming thing is placed at the side of the conduit instead of the end (see pics below) then the device does not receive power. However if the conduit is extended to a length of 2, then the item on the side of the first conduit receives power. 

![image](https://user-images.githubusercontent.com/5566872/49582331-b8d57000-f922-11e8-8905-e6db1447059c.png)
![image](https://user-images.githubusercontent.com/5566872/49582338-bd9a2400-f922-11e8-9477-bfac3462521d.png)
![image](https://user-images.githubusercontent.com/5566872/49582355-cd196d00-f922-11e8-903c-a0d56c644077.png)
![image](https://user-images.githubusercontent.com/5566872/49582365-d30f4e00-f922-11e8-9658-dcc105400834.png)

Additionally it would appear that the EnderIO conduits do not transmit power at all beyond 1 block space from the engine unless the first energy consuming device is "inline" with the engine itself:

![image](https://user-images.githubusercontent.com/5566872/49582489-38633f00-f923-11e8-9276-dd8323288b1a.png)
In this image, the Energy Cell on the right is "inline" with the engine. In this setup power is able to flow to the second energy cell on the left. 

![image](https://user-images.githubusercontent.com/5566872/49582544-68124700-f923-11e8-946f-2af15fb9b532.png)
In this image, the one on the left is now the one that is 1 block space away, however because it is not "inline" with the engine the second energy cell which while it is "inline" with the engine it is beyond 1 block space is not receiving power. I have confirmed that this problem, as well as all of the documented energy problems listed above only exist when working with the Methane Digester Engine. 

As for the second issue that has returned, and that is of the engines pulling power into themselves and acting in a way as a type of energy conduit. 

![image](https://user-images.githubusercontent.com/5566872/49582778-18804b00-f924-11e8-95e5-16e3c90ef27c.png)
In this image I have 3 engines setup in series, with fuel only being supplied to the first engine. There is a single leadstone energy conduit between them. 

![image](https://user-images.githubusercontent.com/5566872/49582838-3ea5eb00-f924-11e8-9523-99b9d3f4b0fa.png)
As you can see, the first one is showing green lights and is receiving fuel and doing work. The other 2 are not receiving fuel and are not doing work. 

![image](https://user-images.githubusercontent.com/5566872/49582874-5b422300-f924-11e8-8fd0-6cad0e5ec45c.png)
This is the GUI from the last engine in the series, which is receiving power from the first one in the series. 

![image](https://user-images.githubusercontent.com/5566872/49582900-71e87a00-f924-11e8-8893-183c0420051e.png)
This is from the second one in the series which is currently receiving power from the first one, and then transmitting that power to the last one.

**Comment by GenDeathrow** [comment]
I'm guess by the way you started the message, a few dev's responses wasn't so nice.  I'll just say is I will try and look into this as soon as I can. Right now I have a lot of projects. But this is a pretty in-depth description of whats happening. All I can think is something has changed with the way forge energy and cofh handles the energy between each other. Hatchery actually uses FE, but when I wrote this.. everyone only really knew RF. So I just labeled it that. Cause it was cross compatible anyways.  But maybe not so much the case atm.
